lib/friendly_shipping/services/ship_engine/parse_void_response.rb

```diff
@@ -6,18 +6,20 @@ module FriendlyShipping
   module Services
     class ShipEngine
       class ParseVoidResponse
-        include Dry::Monads::Result::Mixin
+        extend Dry::Monads::Result::Mixin
 
-        attr_reader :response
-
-        def initialize(response:)
-          @response = response
-        end
-
-        def call
+        def self.call(request:, response:)
           parsed_json = JSON.parse(response.body)
           approved, message = parsed_json["approved"], parsed_json["message"]
-          approved ? Success(message) : Failure(message)
+          if approved
+            Success(
+              ApiResult.new(message, original_request: request, original_response: response)
+            )
+          else
+            Failure(
+              ApiResult.new(message, original_request: request, original_response: response)
+            )
+          end
```

Comment:
How about this? (Unless you anticipate each `ApiResult` will diverge at some point.)

```
result = ApiResult.new(message, original_request: request, original_response: response)
approved ? Success(result) : Failure(result)
```

Reply:
That was actually a bug, the second thing should have been an `ApiFailure` instance.